ansible/roles/test/files/ptftests/vxlan-decap.py

```diff
@@ -0,0 +1,293 @@
+            else:
+                raise Exception("Portchannel '%s' ip address is not found" % name)
+
+            self.pc_info.append((ip, members))
+
+        self.tests = []
+        vni_base = 336
+        src_ip = "8.8.%d.%d"
+        for name, data in graph['minigraph_vlans'].items():
+            test = {}
+            test['name'] = name
+            test['acc_ports'] = [graph['minigraph_port_indices'][member] for member in data['members']]
+            vlan_id = int(name.replace('Vlan', ''))
+            test['vni'] = vni_base + vlan_id
+            test['src_ip'] = src_ip % (vlan_id / 256, vlan_id % 254 + 1)
+
+            gw = None
+            prefixlen = None
+            for d in graph['minigraph_vlan_interfaces']:
+                if d['attachto'] == name:
+                    gw = d['addr']
+                    prefixlen = d['prefixlen']
+                    break
+            else:
+                raise Exception("Vlan '%s' is not found" % name)
+
+            test['vlan_gw'] = gw
+
+            number_of_dataplane_ports = len(self.dataplane.ports)
+            if number_of_dataplane_ports > 256:
+                raise Exception("Too much dataplane ports for the test")
+            if prefixlen > 24:
+                raise Exception("The prefix len size is too small for the test")
+
+            test['vlan_ip_prefix'] = '.'.join(gw.split('.')[0:3])+".%d"
+
+            self.tests.append(test)
+
+        self.dut_mac = graph['dut_mac']
+
+        ip = None
+        for data in graph['minigraph_lo_interfaces']:
+            if data['prefixlen'] == 32:
+                ip = data['addr']
+                break
+        else:
+            raise Exception("ipv4 lo interface not found")
+
+        self.loopback_ip = ip
+
+        self.ptf_mac_addrs = self.readMacs()
+
+        self.generate_ArpResponderConfig()
+
+        self.cmd(["supervisorctl", "start", "arp_responder"])
+
+        self.dataplane.flush()
+
+        return
+
+    def tearDown(self):
+        self.cmd(["supervisorctl", "stop", "arp_responder"])
+
+        return
+
+    def runTest(self):
+        print
+        for test in self.tests:
+            print test['name']
+            res_v = self.Vxlan(test)
+            print "  Vxlan  = ", res_v
+            res_f = self.FromPC(test)
+            print "  FromPC = ", res_f
+            res_t = self.ToPC(test)
+            print "  ToPC   = ", res_t
+            print
+            if self.vxlan_enabled:
+                self.assertTrue(res_v, "VxlanTest failed")
+            else:
+                self.assertFalse(res_v, "VxlanTest must be disabled")
+            self.assertTrue(res_f, "FromPC test failed")
+            self.assertTrue(res_t, "ToPC test failed")
+
+    def Vxlan(self, test):
+        rv = True
+        for n in self.net_ports:
+            for a in test['acc_ports']:
+                res = self.checkVxlan(a, n, test)
+                rv = rv and res
+
+        return rv
+
+    def FromPC(self, test):
```

Comment:
Not sure if `FromPC `and `ToPC` are descriptive, can we use a better name? Something like "FromNetworkHost" and "ToNetworkHost" ? Just a suggestion

Reply:
For me FromPC and ToPC means FromPortChannel and ToPortChannel. FromNetworkHost and ToNetworkHost are less descriptive